hypothesis-python/src/hypothesis/strategies/_internal/types.py

```diff
@@ -332,10 +347,44 @@ def from_typing_type(thing):
             # Reversible is somehow a subclass of Hashable, so we tuplize it.
             # See also the discussion at https://bugs.python.org/issue39046
             typing.Reversible: st.lists(st.integers()).map(tuple),  # type: ignore
-            typing.SupportsAbs: st.complex_numbers(),
-            typing.SupportsComplex: st.complex_numbers(),
-            typing.SupportsFloat: st.floats(),
-            typing.SupportsInt: st.floats(),
+            typing.SupportsAbs: (
```

Comment:
I'm not super happy about adding more stuff into this dictionary like this... particularly with all the try accepts and stuff below. It can be quite a pain to debug. 

I'm happy to separate the declaration into wrapper functions if that would help....  but i'm never quite sure where to put stuff in this module.

any plans for a cleanup in here after v5.0 ? would be happy to help...